js/main.js

```diff
@@ -0,0 +1,37 @@
+
+var $avatarUrl = document.querySelector('.avatarUrl');
+var $pfpPreview = document.querySelector('.pfp');
+
+$avatarUrl.addEventListener('input', function (e) {
+  $pfpPreview.setAttribute('src', e.target.value);
+});
+
+var $form = document.querySelector('form');
+var previousEntries = localStorage.getItem('code-journal-local-storage');
+
+if (previousEntries) {
+  data.entries = JSON.parse(previousEntries);
+}
+
+$form.addEventListener('submit', function (e) {
+  e.preventDefault();
+  data.profile.avatarUrl = $form.elements.avatarUrl.value;
+  data.profile.username = $form.elements.username.value;
+  data.profile.fullName = $form.elements.fullname.value;
+  data.profile.location = $form.elements.location.value;
+  data.profile.bio = $form.elements.bio.value;
+
+  data.entries.push(data.profile);
+
+  $form.elements.avatarUrl.value = '';
```

Comment:
Resetting your form can be done more succinctly:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reset